SQL.js

```diff
@@ -17,28 +17,18 @@ class SqlStatement {
 
   glue (pieces, separator) {
     const result = { strings: [], values: [] }
-
     let carryover
     for (let i = 0; i < pieces.length; i++) {
       const strings = Array.from(pieces[i].strings)
-      if (typeof carryover === 'string') {
+      if (i > 0) {
         strings[0] = carryover + separator + strings[0]
-        carryover = null
-      }
-
-      if (strings.length > pieces[i].values.length) {
-        carryover = strings.splice(-1)[0]
       }
+      carryover = strings.splice(-1)[0]
       result.strings.push.apply(result.strings, strings)
       result.values.push.apply(result.values, pieces[i]._values)
     }
-    if (typeof carryover === 'string') {
-      result.strings.push(carryover)
-    }
 
-    if (result.strings.length === result.values.length) {
-      result.strings.push('')
-    }
```

Comment:
so this logic was basically unnecessary/wrong?

Reply:
Yes, because the separator is only added when strings.length > pieces[I].values.length, with this solution we always add the separator and the last char from the last statement (carryover+separator).

Because of this, the old line 35 is not required because carryover is always a string.
And the old line 39 is not covered by any of 98 tests, and I couldn't reproduce any use case that uses this logic.